library/kani/src/futures.rs

```diff
@@ -41,3 +44,216 @@ const NOOP_RAW_WAKER: RawWaker = {
 
     RawWaker::new(std::ptr::null(), &RawWakerVTable::new(clone_waker, noop, noop, noop))
 };
+
+static mut EXECUTOR: Scheduler = Scheduler::new();
+const MAX_TASKS: usize = 16;
+
+type BoxFuture = Pin<Box<dyn Future<Output = ()> + Sync + 'static>>;
+
+/// Allows to parameterize how the scheduler picks the next task to poll in `spawnable_block_on`
+pub trait SchedulingStrategy {
```

Comment:
Can you please improve the documentation? How would a user implement this?

Reply:
Could you be more specific as to what information is missing? Only one method has to be implemented and its documentation should include all the user needs to know?

Reply:
Also, I think the Scheduler is `pub(crate)`, so we shouldn't expect users to know about it in the first place. You should document this trait as to what this does, not how. Something like:

```rust
/// Trait that determines the possible sequence of tasks scheduling for a  harness.
/// 
/// <Expand what this means for the verification task at hand. Avoid giving implementation details>
///
/// <Optional Example>
``

Reply:
Sounds good. Shouldn't this trait be `pub(crate)` though?